hardware/national_instruments_x_series.py

```diff
@@ -141,6 +140,10 @@ def on_activate(self):
         self._oversampling = 0
         self._lock_in_active = False
 
+        self._photon_sources = self._photon_sources if self._photon_sources is not None else list()
+        self._scanner_counter_channels = self._scanner_counter_channels if self._scanner_counter_channels is not None else list()
+        self._scanner_ai_channels = self._scanner_ai_channels if self._scanner_ai_channels is not None else list()
+
```

Comment:
default value is an empty list, so why check for `None`? The user would have to explicitly declare `None` in the config for this to trigger.

Reply:
I can answer this one: `None` can occur when not adding any list elements after the key word in the config file.
As soon as there is a new line with a `-`, a list is returned.

Reply:
It's just extra code, but ok... in the unlikely case someone declares a config option without giving a value this specific module will cope with it... ;)

By the way: This piece of code would also be rendered obsolete if checks for empty channel lists would be done by converting it to `bool` instead of checking `len == 0`. (see below)